pkg/controller/operators/olm/operator.go

```diff
@@ -310,6 +314,32 @@ func (a *Operator) syncObject(obj interface{}) (syncError error) {
 		a.requeueOwnerCSVs(metaObj)
 	}
 
+	namespace, ok := obj.(*corev1.Namespace)
```

Comment:
I think this may be too aggressive for resyncing.

The cases where we know we need to resync:
 - namespace is added/deleted
 - namespace's labels updated

This makes me think we may want to detect those cases directly by passing in specific handlers for those events.

Reply:
If we're looking at all three add/update/delete, is having separate handlers for each better? Or perhaps you're referring to something more granular that I don't know about?

Reply:
the thing I really want to omit from processing is the case where the namespace is synced periodically, or when there has been some non-label-related change to a namespace

Reply:
If no namespace processing is done then the scenario I described in https://github.com/operator-framework/operator-lifecycle-manager/pull/675#discussion_r249543359 will only eventually work, not instantly as I was hoping could be the case. Is that acceptable?